tests/cypress/scripts/cliHelper.js

```diff
@@ -3,36 +3,221 @@
+ * @param {array} state Targeted state to reset.
+ * @param {object} options Additional options for resetting the new resources state.
+ */
+export const resetNewResourceState = (state, options = {}) => {
+  Cypress.env(state.kind, false)
+}
+
+/**
+ * Reset the state of the kind objects that are required by the test environment.
+ * @param {array} state Targeted state to reset.
+ * @param {object} options Additional options for resetting the new resources state.
+ */
+export const resetNewMultiResourceState = (state, options = {}) => {
+  state.forEach((s) => {
+    Cypress.env(s.kind, false)
+  })
+}
+
+/**
+ * Helper tool for managing resources created by the `oc` cli command.
+ */
 export const cliHelper = {
-  checkIfLoggedIn: () => {
-    cy.url().then((res) => {
-      if (res.includes('oauth-openshift')) {
+  /**
+   * Create new instance of the resource object within the test cluster environment.
+   * @param {object} resource The resource object to create. (Supported: application, deployment, namespace)
+   * @param {string} kubeconfig The kubeconfig file to create the resource object. (Only required for managed cluster testing)
+   * @param {object} options Additional options for creating the new resources object.
+   */
+  createResource: (resource, kubeconfig = '', options = {}) => {
+    // Build command line arguments for the resource creation.
+    var cmd = `${kubeconfig} oc get ${resource.kind} ${resource.name}`
+
+    // Check to see if the namespace was passed in the object.
+    if (resource.namespace) {
+      cmd += ` -n ${resource.namespace}`
+    }
+
+    cy.exec(cmd, { failOnNonZeroExit: false }).then((res) => {
+      if (!res.stderr) {
+        cy.log(
+          `${resource.kind}: ${resource.name} exist within the current cluster environment.`
+        )
+        Cypress.env(resource.kind, true)
+      } else {
+        cy.log(
+          `${resource.kind}: ${resource.name} does not exist within the cluster environment. Preparing to create ${resource.kind} resource.`
+        )
+        Cypress.env(resource.kind, false)
+      }
+
+      if (!Cypress.env(resource.kind)) {
+        cmd = cmd.replace('oc get', 'oc create')
+
+        // Check to see if the image was passed in the object.
+        if (resource.image) {
+          cmd += ` --image ${resource.image}`
+        }
+
+        if (resource.kind == 'application') {
+          cy.readFile('tests/cypress/templates/application.yaml').then(
+            (cfg) => {
+              let b64Cfg = btoa(
+                cfg
+                  .replaceAll('APPNAME', resource.name)
+                  .replaceAll('NAMESPACE', resource.namespace)
+              )
+              cy.exec(`echo ${b64Cfg} | base64 -d | oc apply -f -`).then(() => {
+                Cypress.env(resource.kind, true)
+              })
+            }
+          )
+        } else {
+          cy.exec(cmd).then(() => {
+            Cypress.env(resource.kind, true)
+          })
```

Comment:
What happens if we fail executing the command? We should at least log the error.

Reply:
If the executing command fails, it will fail the test and attempt to retry the test. It will fail since we are not using the `{ failOnNonZeroExit: false }` option. The failure will be logged within the result files.